### ISSUE TITLE
Add Document to a Folder by Right-Clicking on Folder Name

### ISSUE THREAD

**Issue by novakwh** [vkbo/novelWriter#1519]
Not sure if this has been discussed and already rejected as impossible, but it would be helpful if the menu that opens when you right-click on a folder name also included the option to add a document to that folder. 

Currently, I have to click on a folder name, then go up to the little plus sign to add a document to the folder (and the system seems to "know" what types of document can be added into the selected folder). But my instinct, for whatever reason, is to right-click on the folder name and look for the add document option in the menu that pops up there. 

I am constantly right-clicking on the folder name from force of habit, then I realize that the add document choice is not there, and then I have to go up to the plus sign to add the document. If there's a simple way to provide this option in the right-click menu on a folder name, it would be great! Thanks for considering. (I am using the current release candidate, by the way, and no issues so far.)

**Comment by vkbo** [comment]
It's not impossible. The reason they're on a separate button is so they're more easily reachable by various means, including both button clicks and keyboard shortcut. You can activate the + menu by pressing Ctrl+N on an item.

The reason it was moved up there is because the number of options available were expanded a great deal, which would generate multiple levels of context menus if kept under the right-click approach. It is also meant to be extended at some point with user-defined templates.

So, basically it's a compromise between right-click oriented use and keyboard shortcut oriented use.

**Comment by anthonyirwin82** [comment]
I think it would be nice to be able to add a document or a folder at the location you are right-clicking on.

**Comment by vkbo** [comment]
I think mapping the first level of content from the `+` menu is a good compromise. The rest can be added from the old location. So, basically the add document options, and regular folder.

**Comment by vkbo** [comment]
> I think a keyboard shortcut like ctrl+n for a new document and ctrl+shift+n for a new folder would be nice as well.

Pressing `Ctrl+N` activates the `+` menu already, so this will not be remapped. Pressing `Ctrl+.` will activate the context menu on the selected item.

**Comment by vkbo** [comment]
The exact same actions on the first level of the "Add Item" menu will be available on right-click from version 2.3 and on.

![image](https://github.com/vkbo/novelWriter/assets/1619840/6a13129b-47bf-4604-b3b4-cfe4bb51546e)
